lib/kubernetes-deploy/kubernetes_resource/horizontal_pod_autoscaler.rb

```diff
@@ -0,0 +1,25 @@
+# frozen_string_literal: true
+module KubernetesDeploy
+  class HorizontalPodAutoscaler < KubernetesResource
+    PRUNABLE = true
+    TIMEOUT = 30.seconds
+
+    def deploy_succeeded?
+      conditions = @instance_data.dig("status", "conditions") || []
+      able_to_scale = conditions.detect { |c| c["type"] == "AbleToScale" } || {}
+      able_to_scale["status"] == "True"
+    end
+
+    def deploy_failed?
+      !exists?
+    end
+
+    def timeout_message
+      UNUSUAL_FAILURE_MESSAGE
+    end
+
+    def type
+      'hpa.v2beta1.autoscaling'
```

Comment:
Why this? Don't we want it called `HorizontalPodAutoscaler`?

Reply:
We do, but this method determines how kubectl fetches the resource. And we don't get the status conditions until v2beta1. Do we want to add a new method?